cfn-resources/cluster/cmd/resource/resource.go

```diff
@@ -23,446 +41,261 @@ func castNO64(i *int64) *int {
 	x := cast.ToInt(&i)
 	return &x
 }
+
 func cast64(i *int) *int64 {
 	x := cast.ToInt64(&i)
 	return &x
 }
 
+// validateModel inputs based on the method
+func validateModel(fields []string, model *Model) *handler.ProgressEvent {
+	return validator.ValidateModel(fields, model)
+}
+
 // Create handles the Create event from the Cloudformation service.
 func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
 	setup()
-	_, _ = logger.Debugf("Create() currentModel:%+v", currentModel)
-	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
-	if err != nil {
-		_, _ = logger.Warnf("Create - error: %+v", err)
-		return handler.ProgressEvent{
-			OperationStatus:  handler.Failed,
-			Message:          err.Error(),
-			HandlerErrorCode: cloudformation.HandlerErrorCodeInvalidRequest}, nil
+	_, _ = log.Debugf("Create cluster model : %+v", currentModel)
```

Comment:
do we need all this Debug logs? ill just try to keep only the ones that log error cases

Reply:
also i would try not to log the current model, i has the private an public key

Reply:
debug can have any kind of data, since we publish with only warning option